pkg/reconciler/route/reconcile_resources.go

```diff
@@ -69,27 +69,65 @@ func (c *Reconciler) getClusterIngressForRoute(route *v1alpha1.Route) (*netv1alp
 	return ingresses[0], nil
 }
 
+func (c *Reconciler) getIngressForRoute(route *v1alpha1.Route) (*netv1alpha1.Ingress, error) {
+	// First, look up the fixed name.
+	ingressName := resourcenames.Ingress(route)
+	ingress, err := c.ingressLister.Ingresses(route.Namespace).Get(ingressName)
+	if err == nil {
+		return ingress, nil
+	}
+
+	// If that isn't found, then fallback on the legacy selector-based approach.
```

Comment:
I'm pretty sure there's no legacy methods needed for the all-new ingress? I think all code below this is dead.